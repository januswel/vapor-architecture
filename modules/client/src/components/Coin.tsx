import * as React from 'react'
import { StyleSheet, Text, TouchableOpacity, View } from 'react-native'

const styles = StyleSheet.create({
  container: {
    justifyContent: 'center',
    alignItems: 'center',
    width: 96,
    height: 96,
    backgroundColor: 'silver',
    borderRadius: 48,
  },
  label: {
    fontSize: 32,
  },
})

export interface Actions {
  charge: (price: number) => void
}
export interface Props {
  price: number
  actions: Actions
}
export default (props: Props) => (
  <TouchableOpacity
    onPress={() => {
      props.actions.charge(props.price)
    }}
  >
    <View style={styles.container}>
      <Text style={styles.label}>{props.price}</Text>
    </View>
  </TouchableOpacity>
)
