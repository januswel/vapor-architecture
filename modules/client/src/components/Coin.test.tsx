import * as React from 'react'
import * as renderer from 'react-test-renderer'

import Coin from './Coin'

describe('Coin', () => {
  it('renders coin', () => {
    const component = renderer.create(<Coin price={10} actions={{ charge: () => {} }} />)
    expect(component).toMatchSnapshot()
  })
})
