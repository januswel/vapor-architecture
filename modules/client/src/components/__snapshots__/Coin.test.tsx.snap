// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coin renders coin 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "silver",
        "borderRadius": 48,
        "height": 96,
        "justifyContent": "center",
        "width": 96,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 32,
        }
      }
    >
      10
    </Text>
  </View>
</View>
`;
